solution2.py

```diff
@@ -0,0 +1,22 @@
+# https://leetcode.com/problems/house-robber/submissions/
+
+# // Time Complexity : o(number of houses)
+# // Space Complexity : o(1) = only two choices (to rob, or to not to rob)
```

Comment:
The space should also be O(n), the table is n*2 dimension. 2 is a constant and can be ignored but n (len(nums)is unknown and should be accounted for!